storybookjs/storybook
# CLI: Improve steps in storybook init

## What I did

Switched the order of actions to:
1. install dependencies
2. on success, write storybook config files

Before, if installing deps failed, users would end up with Storybook files, but no dependencies or package scripts. It's better to fail early and only proceed with config files when deps are installed correctly

## How to test

1. Build the CLI
2. Get a project without Storybook, make it contain an invalid entry on package.json
3. Run init, it should fail
4. It should not contain Storybook files

## Checklist



      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

#### Maintainers

      make sure to add the `ci:merged` or `ci:daily` GH label to it.

`["cleanup", "BREAKING CHANGE", "feature request", "bug", "documentation", "maintenance", "dependencies", "other"]`
